File: app/src/main/java/org/qp/android/questopiabundle/GameInterface.java
package org.qp.android.questopiabundle;

import android.net.Uri;

import org.qp.android.questopiabundle.lib.LibRefIRequest;
import org.qp.android.questopiabundle.lib.LibTypeDialog;
import org.qp.android.questopiabundle.lib.LibTypeWindow;

public interface GameInterface {

    boolean isPlayingFile(String filePath);

    void closeAllFiles();

    void closeFile(String filePath);

    void playFile(String path, int volume);

    void doChangeCurrGameDir(Uri newGameDirUri);

    void doRefresh(LibRefIRequest request);

    LibDialogRetValue showLibDialog(LibTypeDialog dialog, String inputString);

    void changeVisWindow(LibTypeWindow type, boolean show);

    /**
     * Set the counter location processing interval to <code>delayMillis</code> milliseconds.
     */
    void setCountInter(int delayMillis);

    /**
     * Execute <code>runnable</code> without processing the location counter.
     */
    void doWithCounterDisabled(Runnable runnable);

}

File: app/src/main/java/org/qp/android/questopiabundle/GameInterface.kt
package org.qp.android.questopiabundle

import org.qp.android.questopiabundle.lib.LibRefIRequest
import org.qp.android.questopiabundle.lib.LibTypeDialog
import org.qp.android.questopiabundle.lib.LibTypeWindow

interface GameInterface {
    fun requestReceiveFile(filePath: String)
    fun requestCreateFile(path: String, mimeType: String)
    fun isPlayingFile(filePath: String)
    fun closeAllFiles()
    fun closeFile(filePath: String?)
    fun playFile(path: String?, volume: Int)
    fun changeGameDir(filePath: String)
    fun doUpdateState(request: LibRefIRequest)
    fun showLibDialog(dialog: LibTypeDialog?, inputString: String?)
    fun changeVisWindow(type: LibTypeWindow?, show: Boolean)
    /**
     * Set the counter location processing interval to `delayMillis` milliseconds.
     */
    fun setCountInter(delayMillis: Int)
    /**
     * Execute `runnable` without processing the location counter.
     */
    fun doWithCounterDisabled(runnable: Runnable?)
}
